operator-framework/operator-sdk-samples
# Upgrade version + few improvements

## Motivation 
- https://github.com/operator-framework/operator-sdk-samples/issues/68 
- https://github.com/operator-framework/operator-sdk/issues/1830

required for the following as well:

- https://github.com/operator-framework/getting-started/issues/43
- https://github.com/operator-framework/getting-started/issues/44

## What
- Upgrade go example from 0.6 to operator-sdk version: v0.10.0, commit: ff80b17737a6a0aade663e4827e8af3ab5a21170
- Add makefile in order to be easier to test the example and clarifies how to use it
- Add the new version of olm-catalogue files 
- Change the Readme

## Steps to verify
- Build an image with `operator-sdk build `
- Run `make install` and check that the Memcached pods are deployed and running 

Following the local tests performed with the image `cmacedo/memcached-operator:operator-sdk-samples` 

<img width="1664" alt="Screenshot 2019-09-05 at 23 13 03" src="https://user-images.githubusercontent.com/7708031/64388722-e1b49e00-d037-11e9-88ef-194d6ce46678.png">
